File: app/src/main/java/com/tregula/graedukacyjna/domain/concurrent/EducationGameSchedulers.java
package com.tregula.graedukacyjna.domain.concurrent;

import io.reactivex.Scheduler;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

public class EducationGameSchedulers implements GameSchedulers {

    @Override
    public Scheduler main() {
        return AndroidSchedulers.mainThread();
    }

    @Override
    public Scheduler io() {
        return Schedulers.io();
    }
}

File: app/src/main/java/com/tregula/graedukacyjna/concurrent/EducationGameSchedulers.kt
package com.tregula.graedukacyjna.concurrent

import io.reactivex.Scheduler
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers
import javax.inject.Inject

class EducationGameSchedulers @Inject constructor() : GameSchedulers {

    override fun main(): Scheduler = AndroidSchedulers.mainThread()

    override fun io(): Scheduler = Schedulers.io()
}
